helpers/metadata.go

```diff
@@ -33,3 +38,37 @@ func HasLabelWithValue(object client.Object, label, value string) bool {
 
 	return false
 }
+
+// CopyLabelsByPrefix copies all labels from a source object to a destination object where the key matches the specified prefix.
+// If replacementPrefix is different from prefix, the prefix will be replaced while performing the copy.
+func CopyLabelsByPrefix(src *metav1.ObjectMeta, dest *metav1.ObjectMeta, prefix string, replacementPrefix string) {
```

Comment:
Very nice!

One could potentially simplify the function signature, like this:
`func CopyLabelsByPrefix(src , dest *metav1.ObjectMeta, prefix, replacementPrefix string) {`

Reply:
Done.